### ISSUE TITLE
Restrict props types

### ISSUE THREAD

**Issue by jgoux** [tw-in-js/twind-react#4]
Hello,

When using `styled` I noticed that all the possible html props types are available on the component created which hid the variant in a very long list.

Is there a way to only display the component's variants + a specific subset (or no subset at all) of the possible other native props?

For example I defined a `Text` component using `styled("span", { /* some variants */ })` and here is what I have as possible props (a very long list ^^) : 
![image](https://user-images.githubusercontent.com/1443499/112496154-cc6fa380-8d84-11eb-883c-185f802f0f04.png)

**Comment by sastan** [comment]
That kind of autocompletion is not helpful. I noticed that too. Do you know how emotion/styled-components handle that?

I see 2 ways to solve this:

1. remove all additional attributes and use `Record<string, unknown>` – that would remove all type checking for these props
2. Extend the Twind VS Code Extension to show the variants first in the list

Did I miss something?
Are there other ways?

**Comment by jgoux** [comment]
This doesn't seem to be an issue with stitches : 

![image](https://user-images.githubusercontent.com/1443499/112652275-85011a00-8e4d-11eb-85e0-4152adccf93c.png)

Codesandbox here : https://codesandbox.io/s/competent-almeida-oxf56?file=/src/App.tsx

They are doing a lot of heavy lifting at the type level : https://github.com/modulz/stitches/blob/canary/packages/react/types/index.d.ts#L42

**Comment by sastan** [comment]
I see the same problem with stitches in the codesandbox:  https://codesandbox.io/s/busy-matsumoto-tp3ik?file=/src/App.tsx

![Bildschirmfoto 2021-03-26 um 16 40 32](https://user-images.githubusercontent.com/514405/112656900-25593d80-8e52-11eb-9529-df7b9d26ad88.png)
![Bildschirmfoto 2021-03-26 um 16 40 48](https://user-images.githubusercontent.com/514405/112656907-27230100-8e52-11eb-848b-1f20989e4aa0.png)

**Comment by sastan** [comment]
**But that doesn't mean it can be improved**

Maybe we go with solution 2: Extend the Twind VS Code Extension to show the variants first in the list

What do you think?

**Comment by jgoux** [comment]
Yes it would be a great improvement to start with, then maybe in the future we will find TS tricks to have a more native solution. 👍

**Comment by sastan** [comment]
I believe the _problem_ is that the default typescript IntelliSense shows all possible properties. But if you remove them you either allow any properties or no additional properties.

I'll try to take a look at the weekend. 

Would be ok with if I move this to the typescript-plugin repo?

**Comment by jgoux** [comment]
Of course @sastan, let's move this and have a great week-end!
